### DISCUSSION TITLE
Subtitles for videos

### DISCUSSION THREAD

**Discussion by sircharlo** [sircharlo/meeting-media-manager#666] [Ideas] [answered]
I came across an interesting scenario recently during a meeting of a congregation that hosts a foreign language group. For the benefit of that group, every week they painstakingly, manually add the subtitle tracks of the videos that have them, to the videos in the host congregation's language.

This struck me as a scenario that might be feasible in M3. 
It would require an additional settings toggle to "overlay subtitles in secondary language when available", along with a second language picker.

When downloading MW meeting videos, if the `subtitles` property is not empty in the `GETPUBMEDIALINKS` call for that video ([example](https://b.jw-cdn.org/apis/pub-media/GETPUBMEDIALINKS?output=json&pub=lffv&track=262&langwritten=E)), then M3 would try to call `GETPUBMEDIALINKS` for the secondary language and get the value of the `subtitles` property there, if it exists, and download the `vtt` file. Then, add the subtitle file as a `<track>` for the corresponding video in the media player while it's being played for the congregation.

What do you think @mtdvlpr ?

**Comment by mtdvlpr** [comment, marked as answer]
For sure! One question. Why check if the video in the primary language has subtitles? Does a video either have subtitles for (almost) every language or for zero languages?

**Reply by sircharlo** [reply]
No, I guess you're right. No reason to not just check the secondary language every time.

**Reply by sircharlo** [reply]
Implemented in https://github.com/sircharlo/meeting-media-manager/pull/670. Thanks @mtdvlpr !

**Comment by mtdvlpr** [comment]
This scenario just popped into my head. What if an old video comes along for language X that is in English but with subtitles for language X inside the video. When we put subtitles for the extra language Y on top of it, it will cause issues. Thoughts on this?

**Reply by mtdvlpr** [reply]
Oh wait, there's a subtitled property that we could check against. Nice!
